Changed all events to include the INuimoController itself as the sender.

Inrego opened the pull request:
It's good practice for events to include a sender object, so the listening object can know which nuimo device actually fired the event (in case of multiple devices). I've added an INuimoController parameter to all events, and call them with this as the parameter. That way, if a class is subscribed to events on multiple devices, it will always know which one fired the event.

Inrego committed:
Changed all events to include the INuimoController itself as the sender.

Inrego committed:
Changed all events to include the INuimoController itself as the sender.

larsblumberg commented:
Awesome! Before merging, can you please update the code example in `README.md` file to reflect this change? Thanks a lot!

Inrego committed:
Updated example in ReadMe

Inrego committed:
Updated example in ReadMe

Inrego commented:
Done :)

larsblumberg commented on README.md:
```diff
@@ -33,69 +33,69 @@ The following code example demonstrates how to list paired Nuimos, connect a Nui
+            Debug.WriteLine("LED matrix displayed");
+        }
     }
-}
```
Thanks for the change – but any reason you indented all code instead of justing adding the `INuimoController controller` argument to event handlers?

Inrego replied:
Oh, oops. I had it through my IDE to make sure it was completely valid code. I'll fix it when I get to my pc again tomorrow afternoon

hansmbakker committed:
Do not dispatch bluetooth events to ui thread

- Do not dispatch bluetooth events to UI thread to support headless apps, i.e. IoT Core apps
- Increase timeout for notification subscriptions to 5 seconds

larsblumberg commented:
Dear @Inrego please rebase your PR's branch on top of master as we've removed dispatching to the UI thread.

hansmbakker commented:
@Inrego thanks for this PR!

I did the rebasing in https://github.com/wind-rider/nuimo-windows/tree/event-sender. Can you check it and push it to your own branch so that this PR can be merged? :)

Inrego commented:
I'm not yet a master at git, still learning. I messed up my fork, by doing changes to the master branch and submitting them as a PR: https://github.com/getsenic/nuimo-windows/pull/8

Now I'm not sure how to continue without screwing up that PR :smile:

hansmbakker commented:
What you can do (not sure if this is the perfect way, I'm also still learning):
- review the changes I did to resolve the merge conflicts of your branch
- add [my repo](https://github.com/wind-rider/nuimo-windows) with alias "wind-rider"
- merge the changes in my branch "[event-sender](https://github.com/wind-rider/nuimo-windows/tree/event-sender)" to your LOCAL branch. This should work since I used your branch as a starting point.
- push the changes from the merge to your ORIGIN branch (the remote on github). This should add the changes to this PR.

If this doesn't work for you then I can make a PR from my branch to your branch.

larsblumberg commented:
@Inrego: I'd suggest to rebase/merge your PR's branch to the master of this repository. Here's a step by step tutorial how to sync your repo to the master of this repository: https://help.github.com/articles/syncing-a-fork/

Before starting with step 1 please make sure to:

> Before you can sync your fork with an upstream repository, you must configure a remote that points to the upstream repository in Git.

as said in that tutorial. Let me know if you have further questions – happy to help.

Inrego committed:
Re-implemented this to event invocations after rebase.

Inrego committed:
Merge

Inrego committed:
Fixed indentation of example in README

Inrego commented:
I think I figured it out. If I'm not mistaken, it should be all good now. You better double-check :smile:

larsblumberg commented on README.md:
```diff
@@ -36,71 +36,68 @@ using System.Diagnostics;
+		_nuimoController.ConnectionStateChanged += OnConnectionState;
 		_nuimoController.BatteryPercentageChanged += OnBatteryPercentage;
-		_nuimoController.LedMatrixDisplayed       += OnLedMatrixDisplayed;
+		_nuimoController.LedMatrixDisplayed += OnLedMatrixDisplayed;
```
Hi @Inrego, sorry for responding so late. Can you please undo the formatting change here? The spaces are kept here intentionally to improve reading.

larsblumberg commented on README.md:
```diff
@@ -36,71 +36,68 @@ using System.Diagnostics;
+								"   . .   " +
+								" ..   .. " +
+								"         " +
+								"         ";
```
Can you please use the same formatting for the matrix string as in the current README file? This said, first matrix string line starts next line (line 80) and all lines start in the same column.

Additionally, the matrix should only be sent if connection state changes to connected. I suggest to use an early exit:

```
private void OnConnectionState(INuimoController controller, NuimoConnectionState nuimoConnectionState)
{
    Debug.WriteLine("Connection state: " + nuimoConnectionState);
    if (nuimoConnectionState != NuimoConnectionState.Connected) { return; }
    //...Display matrix here...
```

Inrego commented:
Since I'll probably end up with the same rebase mess as with my portable library PR, I am going to close this one too, and create a new PR soon. In the same process, I am also completely deleting my fort and re-forking this main repo. I had also created a branch mess over there 😄
